src/third-party/arduino-json-5.2/include/ArduinoJson/Arduino/String.hpp

```diff
@@ -9,11 +9,8 @@
 
 #ifndef ARDUINO
 
-#ifndef ARDUINO_STRING_OVERRIDE

```

Comment:
This change I made by hand to 5.1.1 we will need to make it again unless we can think of any other clever hacks. 
Basically if you are not running on an Arduino board it will try to define the arduino String class for it's local tests. The issue is that we do the same thing for our tests and they will conflict.